transaction/src/manifest/decompiler.rs

```diff
@@ -150,7 +150,7 @@ pub fn decompile_instruction<F: fmt::Write>(
                 f,
                 "TAKE_FROM_WORKTOP_BY_IDS Array<NonFungibleId>({}) ResourceAddress(\"{}\") Bucket(\"{}\");",
                 ids.iter()
-                    .map(|k| format!("NonFungibleId(\"{}\")", hex::encode(&k.to_vec())))
+                    .map(|k| format!("NonFungibleId({})", k))
```

Comment:
I'd rather we didn't rely on the fact that `k`'s display method writes it in manifest-speak, and call an explicit method here (and ditto for other methods).

EG I could imagine we have separate methods for the two different display methodologies from the table:
* `to_manifest_string(&self) -> String` / `from_manifest_string(string: &str) -> Self`
* `to_simple_string(&self) -> String` / `from_simple_string(id_type: NonFungibleIdType, string: &str) -> Self`

And we call `to_manifest_string` here.

I would also rather that display outputs the simple string - because display is intended for users reading, and debug outputs the manifest string - but thoughts welcome.

Separately, we should have the following on NonFungibleAddress:
* `to_canonical_string(&self, bech32_encoder: &Bech32Encoder) -> String` / `from_canonical_string(bech32_decoder: &Bech32Decoder, id_type: NonFungibleIdType, string: &str) -> Self`

Which uses that which encodes the resource address, then `:`, then the simple string representation of the id.

Finally, we can use that to implement `impl<'a> ContextualDisplay<AddressDisplayContext<'a>> for NonFungibleAddress` (see eg `model/component/data.rs` for reference)

Reply:
Removed NonFungibleAddress::from_str() function, in Simulator/publish command switched to String and added code to parse from String to hex, not sure how else this could be done without changing input arguments.